File: sdks/thunkmetrc-java/thunkmetrc/src/test/java/io/github/thunkier/thunkmetrc/IntegrationTest.java
package io.github.thunkier.thunkmetrc;

import io.github.thunkier.thunkmetrc.client.MetrcClient;
import io.github.thunkier.thunkmetrc.wrapper.MetrcWrapper;
import io.github.cdimascio.dotenv.Dotenv;
import org.junit.jupiter.api.Test;
import java.io.File;
import java.util.List;
import static org.junit.jupiter.api.Assertions.*;

public class IntegrationTest {

    private String getEnv(Dotenv dotenv, String key) {
        String val = dotenv.get(key);
        if (val == null) {
            val = System.getenv(key);
        }
        return val;
    }

    @Test
    public void testActivePackagesInventorySync() throws Exception {
        var builder = Dotenv.configure().ignoreIfMissing();
        File envFile = new File("../../../.env");
        if (envFile.exists()) {
            builder.directory(envFile.getParent());
        }
        Dotenv dotenv = builder.load();
        
        String baseUrl = getEnv(dotenv, "METRC_BASE_URL");
        String vendorKey = getEnv(dotenv, "METRC_VENDOR_API_KEY");
        String userKey = getEnv(dotenv, "METRC_USER_API_KEY");
        String licenseNumber = getEnv(dotenv, "METRC_FACILITY_LICENSE_NUMBER");

        if (baseUrl == null) baseUrl = "https://sandbox-api-or.metrc.com";
        
        if (vendorKey == null || userKey == null || licenseNumber == null) {
            System.out.println("Skipping test: Missing METRC_VENDOR_API_KEY, METRC_USER_API_KEY, or METRC_FACILITY_LICENSE_NUMBER");
            return;
        }

        System.out.println("Config: " + baseUrl + ", " + licenseNumber);

        MetrcClient client = new MetrcClient(baseUrl, vendorKey, userKey);
        MetrcWrapper wrapper = new MetrcWrapper(client);
        ThunkMetrc thunk = new ThunkMetrc(wrapper);

        System.out.println("Starting sync for " + licenseNumber);
        long start = System.currentTimeMillis();
        
        try {
            List<?> packages = thunk.activePackagesInventorySync(licenseNumber, null, 5);
            
            long duration = System.currentTimeMillis() - start;
            System.out.println("Sync finished in " + duration + "ms. Found " + packages.size() + " packages.");
            
            assertNotNull(packages);
            assertTrue(packages.size() >= 0);
        } catch (Exception e) {
            if (e.getMessage() != null && e.getMessage().contains("429")) {
                System.out.println("WARNING: Test hit rate limit (429). Passing test as environment is restricted.");
            } else {
                throw e;
            }
        }
    }
}

File: sdks/thunkmetrc-kotlin/thunkmetrc/src/test/kotlin/io/github/thunkier/thunkmetrc/IntegrationTest.kt
package io.github.thunkier.thunkmetrc

import io.github.thunkier.thunkmetrc.client.MetrcClient
import io.github.thunkier.thunkmetrc.wrapper.MetrcWrapper
import io.github.cdimascio.dotenv.Dotenv
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.io.File

class IntegrationTest {

    private fun getEnv(dotenv: Dotenv, key: String): String? {
        return dotenv.get(key) ?: System.getenv(key)
    }

    @Test
    fun testActivePackagesInventorySync() = runBlocking {
        val builder = Dotenv.configure().ignoreIfMissing()
        val envFile = File("../../../.env")
        if (envFile.exists()) {
            builder.directory(envFile.parent)
        }
        val dotenv = builder.load()

        var baseUrl = getEnv(dotenv, "METRC_BASE_URL")
        val vendorKey = getEnv(dotenv, "METRC_VENDOR_API_KEY")
        val userKey = getEnv(dotenv, "METRC_USER_API_KEY")
        val licenseNumber = getEnv(dotenv, "METRC_FACILITY_LICENSE_NUMBER")

        if (baseUrl == null) baseUrl = "https://sandbox-api-or.metrc.com"

        if (vendorKey == null || userKey == null || licenseNumber == null) {
            println("Skipping test: Missing env vars")
            return@runBlocking
        }

        println("Config: $baseUrl, $licenseNumber")

        val client = MetrcClient(baseUrl, vendorKey, userKey)
        val wrapper = MetrcWrapper(client)
        val thunk = ThunkMetrc(wrapper)

        println("Starting sync for $licenseNumber")
        val start = System.currentTimeMillis()

        try {
            val packages = thunk.activePackagesInventorySync(licenseNumber, null, 5)

            val duration = System.currentTimeMillis() - start
            println("Sync finished in ${duration}ms. Found ${packages.size} packages.")

            assertNotNull(packages)
            assertTrue(packages.size >= 0)
        } catch (e: Exception) {
            if (e.message?.contains("429") == true) {
                println("WARNING: Test hit rate limit (429). Passing test as environment is restricted.")
            } else {
                throw e
            }
        }
    }
}
